File: searching/linear_search/linear_search.py
def linear_search(list, item):
    index = 0
    found = False
    while index < len(list) and found is False:
        if list[index] == item:
            found = True
        else:
            index = index + 1

    return found

def linear_search2(list, item):
    for i in list:
        if i == item:
            return True
    return False

list = [1, 3, 5, 3, 5,6, 5, 7, 3, 3,55435,6]

print(linear_search(list, 5))
print(linear_search2(list, 5))

File: searching/linear_search/linear_search.rs
fn linear_search(list: [i32;7], item: i32) -> bool {
    let (mut index, mut found) = (0, false);
    while index < list.len() && found == false {
        if list[index] == item {
            found = true;
        }
        else{index = index + 1}
    }
    return found;
}

fn linear_search2(list: [i32;7], item: i32) -> bool {
    for c in list.iter() {
        if *c == item {
            return true;
        }
    }
    return false;
}

fn main() {
    let list: [i32;7] = [1, 2, 3, 4, 5, 6, 7];
    let result = linear_search2(list, 5);
    println!("{:?}", result);
}